core/shared/src/main/scala/zio/ExitCode.scala

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2017-2020 John A. De Goes and the ZIO Contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio
+
+case class ExitCode(code: Int) extends AnyVal
```

Comment:
```suggestion
final case class ExitCode(code: Int)
```

Reply:
@jdegoes any particular concern with the value class?

Reply:
This is used one time at the exit of the application, so there's no need to use AnyVal, and using AnyVal won't actually work because of the way this is returned from a lambda.

Reply:
Right, just read the cases where it actually allocates. It's indeed useless here.

Reply:
Done